apps/betterangels-backend/notes/tests/test_queries.py

```diff
@@ -192,57 +192,121 @@ def test_notes_paginated_query(self) -> None:
+            ("org_1_case_manager_2", None, True, 1, ["note_3"]),  # CM 2 has one submitted note
+            ("org_1_case_manager_1", "client_user_2", None, 0, []),  # CM 1 has no notes for client 2
+            ("org_1_case_manager_1", "client_user_1", False, 1, ["note"]),  # CM 1 has one unsubmitted note for client 1
+            (None, None, False, 2, ["note", "note_2"]),  # There are two unsubmitted notes
         ],
     )
     def test_notes_query_filter(
         self,
         case_manager_label: Optional[str],
         client_label: Optional[str],
-        search_terms: Optional[str],
         is_submitted: Optional[bool],
         expected_results_count: int,
-        returned_note_label_1: Optional[str],
-        returned_note_label_2: Optional[str],
+        returned_note_labels: list[str],
+    ) -> None:
+        self.graphql_client.force_login(self.org_1_case_manager_2)
+        # self.note is created in the setup block by self.org_1_case_manager_1 for self.client_user_1
+        self.note_2 = self._create_note_fixture({"purpose": "Client 1's Note", "client": self.client_user_1.pk})[
+            "data"
+        ]["createNote"]
+        self.note_3 = self._create_note_fixture({"purpose": "Client 2's Note", "client": self.client_user_2.pk})[
+            "data"
+        ]["createNote"]
+        self._update_note_fixture(
+            {
+                "id": self.note_3["id"],
+                "isSubmitted": True,
+            }
+        )
+
+        query = """
+            query Notes($filters: NoteFilter) {
+                notes: notesPaginated(filters: $filters) {
+                    totalCount
+                    results{
+                        id
+                    }
+                }
+            }
+        """
+
+        filters: dict[str, Any] = {}
+
+        if case_manager_label:
+            filters["createdBy"] = self.user_map[case_manager_label].pk
+
+        if client_label:
+            filters["client"] = getattr(self, client_label).pk
+
+        if is_submitted is not None:
+            filters["isSubmitted"] = is_submitted
+
+        expected_query_count = 4
+        with self.assertNumQueriesWithoutCache(expected_query_count):
+            response = self.execute_graphql(query, variables={"filters": filters})
+
+        self.assertEqual(response["data"]["notes"]["totalCount"], expected_results_count)
+        notes = response["data"]["notes"]["results"]
+
+        for idx, note_label in enumerate(returned_note_labels):
+            self.assertEqual(notes[idx]["id"], getattr(self, note_label)["id"])
+
+    @parametrize(
+        ("teams, expected_results_count, returned_note_labels, expected_query_count"),
+        [
+            ([SelahTeamEnum.WDI_ON_SITE.name, SelahTeamEnum.SLCC_ON_SITE.name], 2, ["note_2", "note_3"], 4),
+            ([SelahTeamEnum.SLCC_ON_SITE.name], 1, ["note_3"], 4),
+            (["invalid team"], 0, [], 1),
```

Comment:
just a suggestion to show that the validation will work even if it's partially valid
```suggestion
            ([SelahTeamEnum.WDI_ON_SITE.name, "invalid team"], 0, [], 1),
```

Reply:
good idea, thanks!